app/controllers/attributes_controller.rb

```diff
@@ -6,20 +6,12 @@ def show
 
     return unless validate_attributes(attribute_names)
 
-    values = attribute_names.each_with_object({}) do |name, values_hash|
-      if user_attributes.stored_locally? name
-        raise NotImplementedError
-      else
-        oauth_response = OidcClient.new.get_attribute(
-          attribute: name,
-          access_token: @govuk_account_session.access_token,
-          refresh_token: @govuk_account_session.refresh_token,
-        )
-        @govuk_account_session.access_token = oauth_response[:access_token]
-        @govuk_account_session.refresh_token = oauth_response[:refresh_token]
-        values_hash[name] = oauth_response[:result]
-      end
-    end
+    local_attributes = attribute_names.select { |name| user_attributes.stored_locally? name }
+    remote_attributes = attribute_names.reject { |name| user_attributes.stored_locally? name }
+
+    raise NotImplementedError if local_attributes.any?
+
+    values = @govuk_account_session.get_remote_attributes(remote_attributes)
```

Comment:
Same as `has_subscription`. We only use `values` once so we could avoid a local variable.

Reply:
Never mind, you already said why this isn't possible in the previous comment.